colly.go

```diff
@@ -1285,10 +1292,9 @@ func (c *Collector) Clone() *Collector {
 
 func (c *Collector) checkRedirectFunc() func(req *http.Request, via []*http.Request) error {
 	return func(req *http.Request, via []*http.Request) error {
-		if !c.isDomainAllowed(req.URL.Hostname()) {
-			return fmt.Errorf("Not following redirect to %s because its not in AllowedDomains", req.URL.Host)
+		if err := c.checkFilters(req.URL.String(), req.URL.Hostname()); err != nil {
+			return err
```

Comment:
I'd preserve "not following redirect to" part, as it's handy to know that it's redirect that failed the check. E.g. `fmt.Errorf("Not following redirect to %q: %w", req.URL, err)`.

(with `%w`, you can still write the test properly, except you'll need to use `errors.Is`)

Reply:
Thanks, corrected now.